lib/gitsh/prompter.rb

```diff
@@ -35,6 +39,14 @@ def prompt_format
       env.fetch('gitsh.prompt') { DEFAULT_FORMAT }
     end
 
+    def shortened_branch_name
+      if branch_name.length > BRANCH_CHAR_LIMIT
+        branch_name[0...BRANCH_CHAR_LIMIT] + "…"
+      else
+        branch_name
+      end
```

Comment:
Though this could be:

```ruby
def shortened_branch_name
  branch_name[0..BRANCH_CHAR_LIMIT] + ellipsis
end
def ellipsis
  if branch_name.length > BRANCH_CHAR_LIMIT
    "..."
  else
    ""
end
```

I think the above is more clear because the conditional has fewer things in it, but it might also be more ugly.